### PR TITLE
Add delete tag command

### LINKED ISSUE #198: Remove tag command
Allow the user to remove tags from a person

### PR DESCRIPTION
**Pull request by Nicholas-Cheng-De-Fei**
Closes #198 

BizBook can now delete a tag from a person by doing the following command

`deleteTag <index> t/<tag>`

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -151,6 +151,22 @@ Examples:
...
+
+- Deletes the `TAG` for the person at the specified `INDEX`.
+  -The index refers to the index number shown in the displayed person list.
+- The index **must be a positive integer** 1, 2, 3, …​
+- The tag **must be a existing tag** on the person.
+
+Examples:
+
+- `deletetag 1 t/friends`
+- `editnotes 2 t/Client`
```
Comment (ITLimJiaWei) [inline docs/UserGuide.md:168]: Should this be deletetag? Typo maybe

**Review by ITLimJiaWei — CHANGES_REQUESTED** [review]
Looks good but a typo in UserGuide

**Review by ITLimJiaWei — APPROVED** [review]
LGTM

File: docs/UserGuide.md [hunk 2]
```diff
@@ -151,6 +151,22 @@ Examples:
...
 - `find Betsy` followed by `delete 1` deletes the 1st person in the results of the `find` command.
 
+### Delete a Tag of an existing contact: `deletetag`
+
+Deletes a tag from a person in the address book.
+
+Format: `deletetag INDEX t/[TAG]`
+
+- Deletes the `TAG` for the person at the specified `INDEX`.
+  -The index refers to the index number shown in the displayed person list.
```
Comment (joeltio) [inline docs/UserGuide.md:161]: This isn't indented properly so there are some formatting errors

File: docs/UserGuide.md [hunk 3]
```diff
@@ -342,6 +358,7 @@ _Details coming soon ..._
 | **Edit**       | `edit INDEX [n/NAME] [p/PHONE_NUMBER] [e/EMAIL] [a/ADDRESS] [t/TAG]…​`<br> e.g.,`edit 2 n/James Lee e/jameslee@example.com`                                            |
 | **Find**       | `find KEYWORD [MORE_KEYWORDS]`<br> e.g., `find James Jake`                                                                                                            |
 | **Delete**     | `delete INDEX`<br> e.g., `delete 3`                                                                                                                                   |
+| **Deletetag**  | `deletetag INDEX t/[TAG]` <br> e.g. `deletetag 1 t/Client`                                                                                                            |
```
Comment (joeltio) [inline docs/UserGuide.md:361]: Actually is there a reason why the name of the action can't be `Delete Tag` with a space?

File: src/main/java/bizbook/logic/commands/DeleteTagCommand.java [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+ */
+public class DeleteTagCommand extends Command {
+
+    public static final String COMMAND_WORD = "deletetag";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Deletes the tag from a specfic person.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_TAG + "TAG]...\n"
+            + "Example: " + COMMAND_WORD + " 1 " + PREFIX_TAG + "BusniessMan";
```
Comment (joeltio) [inline src/main/java/bizbook/logic/commands/DeleteTagCommand.java:34]:
```suggestion
            + "Example: " + COMMAND_WORD + " 1 " + PREFIX_TAG + "BusinessMan";
```

File: src/main/java/bizbook/logic/commands/DeleteTagCommand.java [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+ * Delete a tag from an existing person in the address book.
+ */
+public class DeleteTagCommand extends Command {
+
+    public static final String COMMAND_WORD = "deletetag";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Deletes the tag from a specfic person.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_TAG + "TAG]...\n"
```
Comment (joeltio) [inline src/main/java/bizbook/logic/commands/DeleteTagCommand.java:33]: Square brackets in the syntax imply that it is optional but `TAG` is not an optional parameter.

File: src/test/java/bizbook/logic/commands/DeleteTagCommandTest.java [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+import bizbook.model.UserPrefs;
+import bizbook.model.person.Person;
+import bizbook.model.tag.Tag;
+import bizbook.testutil.PersonBuilder;
+
+public class DeleteTagCommandTest {
+    private Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+
+    @Test
+    public void execute_editNotesCommand_success() throws Exception {
```
Comment (joeltio) [inline src/test/java/bizbook/logic/commands/DeleteTagCommandTest.java:31]: Could you rename this test?

**Review by joeltio — CHANGES_REQUESTED** [review]

File: docs/UserGuide.md (hunk 3 above)
Comment (Nicholas-Cheng-De-Fei) [reply docs/UserGuide.md:361]: image

From what I understand currently the `parseCommand` method us using a class called matcher and is using a function called group which take the first word as the command phrase then tag will be deemed as another input.

Comment (joeltio) [reply docs/UserGuide.md:361]: As in, the Deletetag in the "Action" part of the table could just be the name of the command right? The name of the action is to delete (a) tag. But the command word is `deletetag`

**Review by joeltio — APPROVED** [review]
LGTM

**Pull request merged**